### PR TITLE
ci/Check dependencies

### PR DESCRIPTION
**Pull request by Luni-4**
### Checklist

- [x] Confirm that `run-checks` script has been executed.

### Related Issues/PRs

_Provide links to relevant issues and dependent PRs._

### Changes

This PR adds dependencies checks to `ci` in order to detect:

- Vulnerabilities
- Duplications
- Unused 

### Testing

_Describe how these changes have been tested._

### CODE REVIEW

**Comment by Luni-4** [comment]
@nathanielsimard 

This PR is ready to be reviewed

**Comment by nathanielsimard** [comment]
@Luni-4 We should fix the CI dependencies step before merging!

**Comment by Luni-4** [comment]
@nathanielsimard 

Feel free to change my PR to fix the `burn` dependencies :)

**Comment by Luni-4** [comment]
@nathanielsimard 

It is ready to be reviewed

**Comment by Luni-4** [comment]
@nathanielsimard @antimora 

I have added a files to check licenses but I do not know if I had configured that correctly. Review [cargo.deny](https://github.com/EmbarkStudios/cargo-deny/blob/main/deny.template.toml) options carefully and see if they are right the ones I set up

**Comment by antimora** [comment]
It seems the dependencies is doing its job and failing the CI. We should review the list of licenses and add to an allowable list. There seems to be some odd licenses that I am not familiar with so it might take some time to review.

**Comment by Luni-4** [comment]
@antimora 

Since I do not have the skills to evaluate the right licenses for this project, I'm leaving this PR to you. Feel free to update it and change the code whatever you want, thanks a lot in advance!

**Comment by Luni-4** [comment]
@nathanielsimard and @antimora 

After a brief dialogue with @antimora, I was able to set up and fix the dependency checker. Below a list of the accomplished things:

- Multiple dependencies are detected but they are not going to stop the CI. I have set up a `warn` level instead of an `error` level

- The following licenses are deemed acceptable:
```
    Apache-2.0 WITH LLVM-exception
    Apache-2.0
    BSD-3-Clause
    CC0-1.0
    ISC
    MIT
    MPL-2.0
    OpenSSL
    Unicode-DFS-2016
    Unlicense
    Zlib
```
- Every unused dependency has been removed

**Review by nathanielsimard — APPROVED** [review]
LGTM, waiting for @antimora to review as well before merging

File: .github/workflows/dependencies.yml
Comment (Luni-4) [reply .github/workflows/dependencies.yml:50]: To install faster the binary, reducing the CI time.

**Review by antimora — APPROVED** [review]
Looks good. Lets a give a try!

Thanks, @Luni-4 for your work with this CI enhancements. I feel we have a world class tooling now =)

**Pull request merged**
